.github/workflows/tests-main.yaml

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: tests-main
+on: # at 5:10 UTC every day and on each push to main
+  schedule:
+    - cron: "10 5 * * *"
+  push:
+    branches:
+      - 'main'
+jobs:
+  unit:
+    name: unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.20'
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set up build.env with phony secrets.
+        run: cp build.sample.env build.env
+      - name: Verify no changes from code generation. If this check has failed, run `make generate` and commit the changes.
```

Comment:
Fixed.

Reply:
@hessjcg I meant shorten the name in this file, not `tests.yaml`. The longer name is good for PRs since it notifies outside users if they need to fix something. Not really needed in this file since fingers crossed the `make generate` should never fail in `main`